README.md

```diff
@@ -1,298 +1,433 @@
 # JavaScript Functional Library Project
 
-## Introduction
+## Learning Goals
 
-"Functional programming (FP)" is a style of programming like record-oriented or
-object-oriented programming. It's very popular in languages that ***LOVE***
-functions, like JavaScript.
+- Gain a greater understanding of JavaScript's built-in collection-processing
+  methods
+- Gain a greater understanding of callback functions
 
-Don't get spooked though, we've been guiding you and coaching you all along to
-think in the "FP" mindset.
+## Introduction
 
-Developers learn new paradigms all the time. Today we're going to practice
-learning a new style of programming and implementing it.
+In this lab, you will gain a deeper understanding of JavaScript's built in
+collection-processing methods (`map`, `filter` etc.) by building your own
+implementation of them. You will also have the opportunity to practice using
+callbacks, including calling a callback from within a function, passing a
+callback to a function, and passing data between functions and callbacks.
+
+The programming approach you will be using in this lab is an example of
+***functional programming (FP)***. There is nothing new or different here —
+we've been guiding you all along to think in the "FP" mindset — but you should
+use this as an opportunity to start understanding some important distinctions
+between functional programming and other styles of programming.
+
+A complete understanding of functional programming requires an understanding of
+a number of advanced topics in JavaScript, including **pure functions**, **side
+effects**, and **immutability**. However, at its most basic, FP can be
+understood as a programming style in which data manipulation occurs through
+functions that return the result of the manipulation without modifying the state
+of the original data. This style of programming can be contrasted with
+programming approaches that use **shared state**, in which data is manipulated
+and the results stored in a central location (commonly known as **state**). You
+will learn more about shared state when you get to React in the next phase.
+
+Read through [this blog post about functional programming][FP] before continuing
+with the lab, but don't worry too much if you don't understand everything you
+read. Your goal should be to begin to get a feel for the concepts and for the
+distinction between functional programming and other styles of programming.
 
-First, read up on [FP][], and then come back.
+## Instructions
 
-## Learning Goals
+Listed below are function signatures for each of the functions you will need to
+build. Each signature details what the name, arguments, and return value of the
```

Comment:
`parameters` instead of `argument` here and elsewhere, when discussing the function signature

Reply:
Done